packages/edit-widgets/src/components/keyboard-shortcuts/index.js

```diff
@@ -1,23 +1,55 @@
+/**
+ * External dependencies
+ */
+import { first, last } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
 /**
  * WordPress dependencies
  */
-import { useEffect } from '@wordpress/element';
+import { useEffect, useCallback } from '@wordpress/element';
 import {
 	useShortcut,
 	store as keyboardShortcutsStore,
 } from '@wordpress/keyboard-shortcuts';
-import { useDispatch } from '@wordpress/data';
+import { useSelect, useDispatch } from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
+import { store as blockEditorStore } from '@wordpress/block-editor';
 
 /**
  * Internal dependencies
  */
 import { store as editWidgetsStore } from '../../store';
 
+let selectedClientId;
+
 function KeyboardShortcuts() {
 	const { redo, undo } = useDispatch( 'core' );
 	const { saveEditedWidgetAreas } = useDispatch( editWidgetsStore );
 
+	// Shortcuts Logic
+	const { rootBlocksClientIds } = useSelect( ( select ) => {
+		const {
+			getBlockOrder,
+			getBlockParentsByBlockName,
+			getSelectedBlockClientId,
+		} = select( blockEditorStore );
+		const newSelectedClientId = getSelectedBlockClientId();
+		if ( newSelectedClientId && newSelectedClientId !== selectedClientId ) {
+			selectedClientId = newSelectedClientId;
+		}
+		const [ selectedParentClientId ] = getBlockParentsByBlockName(
+			selectedClientId,
+			'core/widget-area'
+		);
+		return {
+			rootBlocksClientIds: getBlockOrder( selectedParentClientId ),
```

Comment:
Since we only care about the first and the last, maybe try to only expose them here so that we can avoid unnecessary re-renders.